language/move-compiler/src/naming/translate.rs

```diff
@@ -971,25 +986,18 @@ fn exp_(context: &mut Context, e: E::Exp) -> N::Exp {
         EE::Cast(e, t) => NE::Cast(exp(context, *e), type_(context, t)),
         EE::Annotate(e, t) => NE::Annotate(exp(context, *e), type_(context, t)),
 
-        EE::Call(sp!(mloc, ma_), true, tys_opt, rhs) => {
-            use E::ModuleAccess_ as EA;
+        EE::Call(sp!(mloc, E::ModuleAccess_::Name(n)), true, tys_opt, rhs)
+            if n.value.as_str() == N::BuiltinFunction_::ASSERT_MACRO =>
+        {
             use N::BuiltinFunction_ as BF;
-            assert!(tys_opt.is_none(), "ICE macros do not have type arguments");
+            assert!(
+                tys_opt.is_none(),
+                "ICE builtin macros do not have type arguments"
+            );
```

Comment:
This used to be handled in parsing but I think that will go away in this new system?

Reply:
So I think we need a check+error here for non-empty type args?

Reply:
Actually this was not checked in the parser. I turned the ICE into an error message.